packages/blade/src/components/Badge/__tests__/Badge.native.test.tsx

```diff
@@ -0,0 +1,134 @@
+import { Badge } from '../Badge';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';
+import { InfoIcon } from '~components/Icons';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+describe('<Badge />', () => {
+  it('should render Badge with default props', () => {
+    const label = 'Label';
+    const { toJSON } = renderWithTheme(<Badge>{label}</Badge>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should throw an error when no children are passed', () => {
+    try {
+      // @ts-expect-error testing failure case when there is no children passed
+      renderWithTheme(<Badge />);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        expect(error.message).toEqual('[Blade: Badge]: Text as children is required for Badge.');
+      }
+    }
+  });
+  it('should render small size Badge', () => {
+    const label = 'Label';
+    const { toJSON } = renderWithTheme(<Badge size="small">{label}</Badge>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render medium size Badge', () => {
+    const label = 'Label';
+    const { toJSON } = renderWithTheme(<Badge size="medium">{label}</Badge>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render Badge with Icon', () => {
+    const label = 'Label';
+    const { toJSON } = renderWithTheme(<Badge icon={InfoIcon}>{label}</Badge>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render Badge with bold font', () => {
+    const label = 'LABEL';
+    const { toJSON } = renderWithTheme(<Badge fontWeight="bold">{label}</Badge>);
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render low contrast positive variant Badge', () => {
+    const label = 'Label';
+    const { toJSON } = renderWithTheme(
+      <Badge variant="positive" contrast="low">
+        {label}
+      </Badge>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render high contrast positive variant Badge', () => {
+    const label = 'Label';
+    const { toJSON } = renderWithTheme(
+      <Badge variant="positive" contrast="high">
+        {label}
+      </Badge>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render low contrast negative variant Badge', () => {
+    const label = 'Label';
+    const { toJSON } = renderWithTheme(
+      <Badge variant="negative" contrast="low">
+        {label}
+      </Badge>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render high contrast negative variant Badge', () => {
+    const label = 'Label';
+    const { toJSON } = renderWithTheme(
+      <Badge variant="negative" contrast="high">
+        {label}
+      </Badge>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
```

Comment:
@chaitanyadeorukhkar what do you think about using test.each function from jest? 

```ts

  const variants = ['positive', 'negative', 'notice', 'information', 'neutral'];
  test.each(variants)('should render low contrast %p variant Badge', (variant) => {
    const label = 'Label';
    const { container } = renderWithTheme(
      <Badge variant={variant as BadgeProps['variant']} contrast="low">
        {label}
      </Badge>,
    );
    expect(container).toMatchSnapshot();
  });

  test.each(variants)('should render high contrast %p variant Badge', (variant) => {
    const label = 'Label';
    const { container } = renderWithTheme(
      <Badge variant={variant as BadgeProps['variant']} contrast="high">
        {label}
      </Badge>,
    );
    expect(container).toMatchSnapshot();
  });
```

All of the manual permutations can be deleted.

Reply:
Looks interesting. I'll play around with this to see how it works. But I'm guessing this is a non-blocker. I'll move to L2 meanwhile I try this

Reply:
I think we don't need snapshots for all these cases, makes the snapshots huge and difficult to review 😅